File: async-scala-client/src/main/scala/io/swagger/client/model/LegalReason_taxScope.scala
package io.swagger.client.model

import org.joda.time.DateTime
import java.util.UUID


case class LegalReason_taxScope (
  taxType: String,  // This is the Tax Type to apply legal messages. - &#39;NONE&#39; - &#39;INSS&#39; - &#39;IRRF&#39; - &#39;IRPJ&#39; - &#39;PIS&#39; - &#39;COFINS&#39; - &#39;CSLL&#39; - &#39;IPI&#39; - &#39;ICMS&#39; - &#39;II&#39; - &#39;IOF&#39; - &#39;ISS&#39; - &#39;APROXTOTALTAX&#39; 
specializedTaxType: Option[String],  // These are the specific tax types to which a message may be applied - &#39;icms&#39; - &#39;icmsSt&#39; - &#39;icmsStSd&#39; - &#39;icmsPartOwn&#39; - &#39;icmsPartDest&#39; - &#39;icmsDifaFCP&#39; - &#39;icmsDifaDest&#39; - &#39;icmsDifaRemet&#39; - &#39;icmsRf&#39; - &#39;icmsDeson&#39; - &#39;icmsCredsn&#39; - &#39;pis&#39; - &#39;pisSt&#39; - &#39;cofins&#39; - &#39;cofinsSt&#39; - &#39;ipi&#39; - &#39;ipiReturned&#39; - &#39;ii&#39; - &#39;iof&#39; 
jurisdictionType: String,  // Types of jurisdiction - &#39;NONE&#39; - &#39;City&#39; - &#39;State&#39; - &#39;Country&#39; 
jurisdictionIbgeCode: Option[String]  // Code to identify the Jurisdiction
)

File: CsharpDotNet2-client/src/main/CsharpDotNet2/IO/Swagger/Model/LegalReasonTaxScope.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// Filter this message application to specified tax cases. - taxtype - jurisdictionType 
  /// </summary>
  [DataContract]
  public class LegalReasonTaxScope {
    /// <summary>
    /// This is the Tax Type to apply legal messages. - 'NONE' - 'INSS' - 'IRRF' - 'IRPJ' - 'PIS' - 'COFINS' - 'CSLL' - 'IPI' - 'ICMS' - 'II' - 'IOF' - 'ISS' - 'APROXTOTALTAX' 
    /// </summary>
    /// <value>This is the Tax Type to apply legal messages. - 'NONE' - 'INSS' - 'IRRF' - 'IRPJ' - 'PIS' - 'COFINS' - 'CSLL' - 'IPI' - 'ICMS' - 'II' - 'IOF' - 'ISS' - 'APROXTOTALTAX' </value>
    [DataMember(Name="taxType", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "taxType")]
    public string TaxType { get; set; }

    /// <summary>
    /// These are the specific tax types to which a message may be applied - 'icms' - 'icmsSt' - 'icmsStSd' - 'icmsPartOwn' - 'icmsPartDest' - 'icmsDifaFCP' - 'icmsDifaDest' - 'icmsDifaRemet' - 'icmsRf' - 'icmsDeson' - 'icmsCredsn' - 'pis' - 'pisSt' - 'cofins' - 'cofinsSt' - 'ipi' - 'ipiReturned' - 'ii' - 'iof' 
    /// </summary>
    /// <value>These are the specific tax types to which a message may be applied - 'icms' - 'icmsSt' - 'icmsStSd' - 'icmsPartOwn' - 'icmsPartDest' - 'icmsDifaFCP' - 'icmsDifaDest' - 'icmsDifaRemet' - 'icmsRf' - 'icmsDeson' - 'icmsCredsn' - 'pis' - 'pisSt' - 'cofins' - 'cofinsSt' - 'ipi' - 'ipiReturned' - 'ii' - 'iof' </value>
    [DataMember(Name="specializedTaxType", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "specializedTaxType")]
    public string SpecializedTaxType { get; set; }

    /// <summary>
    /// Types of jurisdiction - 'NONE' - 'City' - 'State' - 'Country' 
    /// </summary>
    /// <value>Types of jurisdiction - 'NONE' - 'City' - 'State' - 'Country' </value>
    [DataMember(Name="jurisdictionType", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "jurisdictionType")]
    public string JurisdictionType { get; set; }

    /// <summary>
    /// Code to identify the Jurisdiction
    /// </summary>
    /// <value>Code to identify the Jurisdiction</value>
    [DataMember(Name="jurisdictionIbgeCode", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "jurisdictionIbgeCode")]
    public string JurisdictionIbgeCode { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class LegalReasonTaxScope {\n");
      sb.Append("  TaxType: ").Append(TaxType).Append("\n");
      sb.Append("  SpecializedTaxType: ").Append(SpecializedTaxType).Append("\n");
      sb.Append("  JurisdictionType: ").Append(JurisdictionType).Append("\n");
      sb.Append("  JurisdictionIbgeCode: ").Append(JurisdictionIbgeCode).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
